internal/ethapi/api.go

```diff
@@ -642,11 +642,8 @@ func (s *PublicBlockChainAPI) doCall(ctx context.Context, args CallArgs, blockNr
 	// Setup context so it may be cancelled the call has completed
 	// or, in case of unmetered gas, setup a context with a timeout.
 	var cancel context.CancelFunc
-	if vmCfg.DisableGasMetering {
-		ctx, cancel = context.WithTimeout(ctx, time.Second*5)
-	} else {
-		ctx, cancel = context.WithCancel(ctx)
-	}
+	ctx, cancel = context.WithCancel(ctx)
+
```

Comment:
Please keep the time limit and set gas to `^uint64(0)`.

Reply:
The default gas price is 50M if the user didn't provide any. We can increase that to maybe max uint64/2. https://github.com/ethereum/go-ethereum/issues/16160 reports issues originating from refunds, which I assume originate from specifying max uint64 as ga limit